api/shared/compare/serializers.py

```diff
@@ -76,3 +90,172 @@ class FlagComparisonSerializer(serializers.Serializer):
+                    "content": content,
+                }
+            )
+        return lines
+
+
+class ImpactedFileSerializer(serializers.Serializer):
+    file_name = serializers.SerializerMethodField()
+    base_name = serializers.CharField()
+    head_name = serializers.CharField()
+    is_new_file = serializers.SerializerMethodField()
+    is_renamed_file = serializers.SerializerMethodField()
+    is_deleted_file = serializers.SerializerMethodField()
+    base_coverage = serializers.SerializerMethodField()
+    head_coverage = serializers.SerializerMethodField()
+    patch_coverage = serializers.SerializerMethodField()
+    change_coverage = serializers.SerializerMethodField()
+    misses_count = serializers.SerializerMethodField()
+    hashed_path = serializers.SerializerMethodField()
+    segments = serializers.SerializerMethodField()
+
+    def get_base_coverage(self, impacted_file: ImpactedFile) -> serializers.JSONField:
+        if impacted_file.base_coverage:
+            return dataclasses.asdict(impacted_file.base_coverage)
+
+    def get_head_coverage(self, impacted_file: ImpactedFile) -> serializers.JSONField:
+        if impacted_file.head_coverage:
+            return dataclasses.asdict(impacted_file.head_coverage)
+
+    def get_patch_coverage(self, impacted_file: ImpactedFile) -> serializers.JSONField:
+        if impacted_file.patch_coverage:
+            return dataclasses.asdict(impacted_file.patch_coverage)
+
+    def get_file_name(self, impacted_file: ImpactedFile) -> serializers.CharField:
+        return impacted_file.file_name
+
+    def get_is_new_file(self, impacted_file: ImpactedFile) -> serializers.BooleanField:
+        base_name = impacted_file.base_name
+        head_name = impacted_file.head_name
+        return base_name is None and head_name is not None
+
+    def get_is_renamed_file(
+        self, impacted_file: ImpactedFile
+    ) -> serializers.BooleanField:
+        base_name = impacted_file.base_name
+        head_name = impacted_file.head_name
+        return (
+            base_name is not None and head_name is not None and base_name != head_name
+        )
+
+    def get_is_deleted_file(
+        self, impacted_file: ImpactedFile
+    ) -> serializers.BooleanField:
+        base_name = impacted_file.base_name
+        head_name = impacted_file.head_name
+        return base_name is not None and head_name is None
+
+    def get_change_coverage(
+        self, impacted_file: ImpactedFile
+    ) -> serializers.FloatField:
+        return impacted_file.change_coverage
+
+    def get_misses_count(self, impacted_file: ImpactedFile) -> serializers.IntegerField:
+        return impacted_file.misses_count
+
+    def get_hashed_path(self, impacted_file: ImpactedFile) -> serializers.CharField:
+        path = impacted_file.head_name
+        encoded_path = path.encode()
+        md5_path = hashlib.md5(encoded_path)
+        return md5_path.hexdigest()
+
+    def get_segments(
```

Comment:
One note about the GraphQL API: we don't query for segments by default when querying the impacted files b/c there can be a lot of segments and we're fetching the source code for each file from GitHub.  Might be worth considering a similar design for this public API.  i.e. return a list of impacted files w/ metadata about them and then a separate endpoint where you can fetch the segments for a particular file

Reply:
👍 created a new endpoint: `/api/v2/{service}/{owner_username}/repos/{repo_name}/compare/segments/{file_path}` that fetches for a specific file. Removed getting segments for each impacted file in the impacted_files endpoint